custom_components/goldair_climate/device.py

```diff
@@ -85,7 +87,13 @@ async def async_inferred_type(self):
 
         _LOGGER.debug(f"Inferring device type from cached state: {cached_state}")
         if "5" in cached_state:
-            return CONF_TYPE_DEHUMIDIFIER
+            if "3" in cached_state:
+                if "7" in cached_state:
+                    return CONF_TYPE_GPCV_HEATER
+                else:
+                    return CONF_TYPE_GECO_HEATER
+            else:
+                return CONF_TYPE_DEHUMIDIFIER
```

Comment:
This is a logical ay to extend it, but I feel like from an understandability point of view this is a bit confusing - we have heaters and the dehumidifier in the same statement. I wonder if it could be more like this:
```python
if "5" in cached_state and "3" in cached_state:
    return CONF_TYPE_GPCV_HEATER if "7" in cached_state else CONF_TYPE_GECO_HEATER
if "5" in cached_state and "3" not in cached_state:
    return CONF_TYPE_DEHUMIDIFIER
if "8" in cached_state:
    # etc